src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/CohostOnAutoInsertEndpointTest.cs

```diff
@@ -0,0 +1,275 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Test.Common;
+using Microsoft.CodBeAnalysis.Remote.Razor.AutoInsert;
+using Microsoft.CodeAnalysis.ExternalAccess.Razor;
+using Microsoft.CodeAnalysis.Razor.AutoInsert;
+using Microsoft.CodeAnalysis.Razor.Settings;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.LanguageServices.Razor.LanguageClient.Cohost;
+using Microsoft.VisualStudio.Razor.Settings;
+using Roslyn.Test.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.VisualStudio.Razor.LanguageClient.Cohost;
+
+public class CohostOnAutoInsertEndpointTest(ITestOutputHelper testOutputHelper) : CohostEndpointTestBase(testOutputHelper)
+{
+    [Theory]
+    [InlineData("PageTitle")]
+    [InlineData("div")]
+    [InlineData("text")]
+    public async Task Component_AutoInsertEndTag(string startTag)
+    {
+        var input = $"""
+            This is a Razor document.
+
+            <{startTag}>$$
+
+            The end.
+            """;
+        var output = $"""
+            This is a Razor document.
+
+            <{startTag}>$0</{startTag}>
+
+            The end.
+            """;
+
+        await VerifyOnAutoInsertAsync(input, output, triggerCharacter: ">");
+    }
+
+    [Theory]
+    [InlineData("PageTitle")]
+    [InlineData("div")]
+    [InlineData("text")]
+    public async Task Component_DoNotAutoInsertEndTag_DisabledAutoClosingTags(string startTag)
+    {
+        var input = $"""
+            This is a Razor document.
+
+            <{startTag}>$$
+
+            The end.
+            """;
+        var output = $"""
+            This is a Razor document.
+
+            <{startTag}>
+
+            The end.
+            """;
+
+        await VerifyOnAutoInsertAsync(input, output, triggerCharacter: ">", autoClosingTags: false, expectResult: false);
```

Comment:
Nit: I think it would be clearer to write out the tests as one long multiline method call. Eg:

```
    await VerifyOnAutoInsertAsync(
        input: """
            ...
            """,
        output: """
            ...
            """,
        tirggerCharacter: ">",
        autoClosingTags: false,
        expectResult: false);
```

That way it's easy to scan down the screen to see all the test parameters, than that down for the input and output, and across for the others.